### PR TITLE
Migrate code block test case to playwright

### PR DESCRIPTION
**Pull request by pavanpatil1**
## What?

Migrate[ code-test.js](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/blocks/code.test.js) to its Playwright version.
## Why?

See #38570 for its background.
## How?

See[ MIGRATION.md](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/MIGRATION.md) for migration steps.
## Testing Instructions

Run` npm run test-e2e:playwright test/e2e/specs/editor/blocks/code.spec.js`

### CODE REVIEW

**Comment by talldan** [comment]
Thanks for working on this @pavanpatil1!

Just a few things to fix in this PR, but on the whole looks good. I'll invite @kevin940726 for any further feedback.

Gutenberg uses the `prettier` code formatter, and it looks like the 'Static Analysis' check is failing because the newly added code isn't formatted. You can run `npm run format` to format files. For an individual file I think the command would be something like `npm run format -- test/e2e/specs/editor/blocks/code.spec.js`.

If you're using an editor like VSCode you can also install a `prettier` plugin and configure it format code automatically.

It also looks like there's a snapshot file used by the old test that needs to be deleted - `specs/editor/blocks/__snapshots__/code.test.js.snap`.

File: test/e2e/specs/editor/blocks/code.spec.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+
+test.describe( 'Code', () => {
+	test.beforeEach( async ({pageUtils}) => {
+		await pageUtils.createNewPost();
+	} );
+
+	test( 'can be created by three backticks and enter', async ({page,pageUtils}) => {
+		await pageUtils.clickBlockAppender();
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/code.spec.js]: We usually want to limit the usage of e2e utils if we can inline them directly. If we're just creating a page, I believe we can just click on the appender button directly:

```suggestion
		await page.click( 'role=button[name="Add default block"i]' );
```

The same code has also been used in `copy-cut-paste` tests. This way we don't have to create the `clickBlockAppender` util (unless we really need it in other situations).

**Review by kevin940726 — COMMENTED** [review]
Apart from the minor issues Dan already mentioned, it mostly looks pretty good! I just have a small feedback to suggest inlining the `clickBlockAppender` util if possible.

**Comment by pavanpatil1** [comment]
Hi @kevin940726, I addressed all the shared feedback. However, the CI is failing on Static Analysis (Linting, License, Type checks...) / All (pull_request)`. I followed the steps mentioned here https://github.com/WordPress/gutenberg/pull/40844#issuecomment-1119317835. Formatted the code still it is failing.

File: test/e2e/specs/editor/blocks/code.spec.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		await page.keyboard.type( 'a' );
+
+		await page.keyboard.press( 'Backspace' );
+		await page.keyboard.press( 'Backspace' );
+
+		// Expect code block to be deleted.
+		expect( await editor.getEditedPostContent() ).toBe( '' );
+	} );
+
+	test( 'should paste plain text', async ( { editor, page, pageUtils } ) => {
```
Comment (kevin940726) [inline test/e2e/specs/editor/blocks/code.spec.js]: You can click on the "[Details](https://github.com/WordPress/gutenberg/runs/6552714683?check_suite_focus=true#step:5:137)" link in the checks to see where it went wrong. Here's the relevant error message in the link:

```
[1] /home/runner/work/gutenberg/gutenberg/test/e2e/specs/editor/blocks/code.spec.js
[1]   45:53  error  'page' is defined but never used  no-unused-vars
```

`page` is not used here in this test, so we can omit it here.

**Review by kevin940726 — CHANGES_REQUESTED** [review]

File: test/e2e/specs/editor/blocks/code.spec.js (hunk 2 above)
Comment (kevin940726) [reply test/e2e/specs/editor/blocks/code.spec.js]:
```suggestion
	test( 'should paste plain text', async ( { editor, pageUtils } ) => {
```

**Comment by kevin940726** [comment]
Thank you for working on this and sorry for the late reply 🙇 .

Unfortunately, it seems like this test had already been migrated in https://github.com/WordPress/gutenberg/pull/41136 by @JustinyAhin. Normally, we would want to close the latest one but we forgot about that 😞 . We'll try to pay attention to this kind of problem more in the future. Sorry about it! 🙇 

We're still very grateful for your work, and hope that you can start another PR soon next time ;)

**Comment by JustinyAhin** [comment]
@pavanpatil1 I'm happy to do a pair coding session with you to work on a migrating couple of tests migration if that interests you :).

**Pull request closed without merging**